src/main/java/com/practice/ts/PracticeTwo/controllers/HomeController.java

```diff
@@ -0,0 +1,67 @@
+package com.practice.ts.PracticeTwo.controllers;
+
+import com.practice.ts.PracticeTwo.models.CommentDto;
+import com.practice.ts.PracticeTwo.models.NewsItem;
+import com.practice.ts.PracticeTwo.models.StoryItemDto;
+import com.practice.ts.PracticeTwo.repositories.CommentRepo;
+import com.practice.ts.PracticeTwo.repositories.StoryItemRepo;
+import com.practice.ts.PracticeTwo.services.HackerNewsService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+public class HomeController {
+    @Autowired
+    HackerNewsService hackerNewsService;
+
+    @Autowired
+    private StoryItemRepo storyItemRepo;
+    @Autowired
+    CommentRepo commentRepo;
+
+    @GetMapping("/top-news-items-with-comments")
+    public List<NewsItem> getTopNewsItems() {
+//        Comparator<NewsItem> comparator =
+        var stories = storyItemRepo.getTopNewsItems(10);
+        var newsItems = stories.stream()
+                .map(item -> {
+                    var parentId = item.id;
```

Comment:
Nitpicking here ;) - Kind of a pointless variable. It's usage number is 1 so to simplify the code you could just do:
`List comment = commentRepo.getCommentDtoByParent(item.id).stream().sorted().toList();`

The other thing I would do is make the newsItem fluent so you could do the following:

`return new NewsItem().setComments(comment).setType(item.type).setTitle(item.title);`

This is achievable through using Lomboks `@Accessors(fluent = true, chain = true)` annotation.